mvebu64boot.c

```diff
@@ -317,19 +317,19 @@ static bool transfer_image_file(int fd, const uint8_t *image)
 	const struct main_header *hdr = (const struct main_header *)image;
 	uint8_t seq = 1;
 
-	printf("Sending image prolog... (%u bytes)\n", (unsigned int)le32_to_cpu(&hdr->prolog_size));
+	fprintf(stderr, "Sending image prolog... (%u bytes)\n", (unsigned int)le32_to_cpu(&hdr->prolog_size));
 	if (!xmodem_transfer(fd, &seq, image, le32_to_cpu(&hdr->prolog_size), true))
 		return false;
 
-	printf("Sending image bootloader... (%u bytes)\n", (unsigned int)le32_to_cpu(&hdr->bl_image_size));
-	if (!xmodem_transfer(fd, &seq, image + le32_to_cpu(&hdr->prolog_size), le32_to_cpu(&hdr->bl_image_size), false))
+	fprintf(stderr, "Sending image bootloader... (%u bytes)\n", (unsigned int)le32_to_cpu(&hdr->bl_image_size));
+	if (!xmodem_transfer(fd, &seq, image + le32_to_cpu(&hdr->prolog_size), le32_to_cpu(&hdr->bl_image_size), true))
```

Comment:
This is completely breaking of bootrom protocol because after transferring the last packet, the answer is ACK without without any other intermediate data.

Reply:
This is breaking the protocol. If broken character is received it is handled by protocol as an error and packet has to be re-transmitted. This change just breaks retransmission for the last packet. The only place where bootrom may send something different, not related to the transfer protocol, is after sending last header packet and before receiving acknowledgement of the received header packet.